org.eclipse.xtext/src/org/eclipse/xtext/resource/impl/ResourceDescriptionsBasedContainer.java

```diff
@@ -62,19 +63,25 @@ public IResourceDescription getResourceDescription(URI uri) {
 	
 	@Override
 	public Iterable<IEObjectDescription> getExportedObjects(EClass type, QualifiedName qualifiedName, boolean ignoreCase) {
-		Iterable<IEObjectDescription> unfiltered = getDescriptions().getExportedObjects(type, qualifiedName, ignoreCase);
+		Iterable<IResourceDescription> resourceDescriptions = filterByURI(getDescriptions());
+		Iterable<IEObjectDescription> unfiltered = IterableExtensions.flatMap(resourceDescriptions, desc -> desc.getExportedObjects(type, qualifiedName, ignoreCase));
```

Comment:
`resourceDescriptions.getExportedObjects(.., name, ..)` is usually very fast. Here I'd opt for filtering after-the-fact, e.g. based on the few found `IEObjectDescriptions`.

Reply:
you are right that this is usually very fast. i just thought it might get confusing if this method would not apply the filter. so if a API user just overrides the resourceDescription filter he might get a result here that would not apply that filter